pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -261,9 +261,16 @@ func (r *reconciler) ensureLoadBalancerService(ci *operatorv1.IngressController,
 		return false, nil, err
 	}
 
+	// BZ2054200: Don't modify/delete services that are not directly owned by this controller
+	ownLBS := isServiceOwnedByIngressController(currentLBService, ci)
+
 	switch {
 	case !wantLBS && !haveLBS:
 		return false, nil, nil
+	case !wantLBS && haveLBS && !ownLBS:
+		return false, nil, fmt.Errorf("refusing to delete load balancer service not owned by ingress controller: %s", controller.LoadBalancerServiceName(ci))
+	case wantLBS && haveLBS && !ownLBS:
+		return false, nil, fmt.Errorf("refusing to update load balancer service not owned by ingress controller: %s", controller.LoadBalancerServiceName(ci))
```

Comment:
The switch statement follows a pattern that appears throughout cluster-ingress-operator and cluster-dns-operator, and I'd prefer we stick to that pattern for consistency.  Could we instead modify the existing cases as follows?
```go
	case !wantLBS && haveLBS:
		if !ownLBS {
			return false, nil, fmt.Errorf("refusing to delete load balancer service not owned by ingress controller: %s", controller.LoadBalancerServiceName(ci))
		}
		// [existing logic]
	case wantLBS && !haveLBS:
		// [existing logic]
	case wantLBS && haveLBS:
		if !ownLBS {
			return false, nil, fmt.Errorf("refusing to update load balancer service not owned by ingress controller: %s", controller.LoadBalancerServiceName(ci))
		}
		// [existing logic]
```

Reply:
By the way, what actually happens with this logic when the service exists but isn't owned by the operator?  If `ensureLoadBalancerService` returns an error, I believe the reconciliation loop will retry, correct?  Does the reconciliation loop retry a bunch of times with backoff?

Reply:
Ah yea that's fair. I kept toying over the best way to keep logic consistent, but I agree with your suggestion.

As best as I can see, if `ensureLoadBalancerService` returns an error, it will retry if ALL errors in list are type `retryable.Error`. However, if there is one error that is not, it doesn't retry. This logic seems funny to me. So our error that we are returning won't make it constantly retry. In fact, it will prevent it from retrying if there is another `retryable.Error` in the error list. See:
https://github.com/openshift/cluster-ingress-operator/blob/542124811a01039852d8b0846da91d33665037f8/pkg/util/retryableerror/retryableerror.go#L40

Reply:
Done

Reply:
> As best as I can see, if `ensureLoadBalancerService` returns an error, it will retry if ALL errors in list are type `retryable.Error`. However, if there is one error that is not, it doesn't retry. 

That isn't completely accurate.  The `Reconciler` interface has nonintuitive semantics:

```go
type Reconciler interface {
	// Reconcile performs a full reconciliation for the object referred to by the Request.
	// The Controller will requeue the Request to be processed again if an error is non-nil or
	// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
	Reconcile(context.Context, Request) (Result, error)
}
```
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler

```go
type Result struct {
	// Requeue tells the Controller to requeue the reconcile key.  Defaults to false.
	Requeue bool

	// RequeueAfter if greater than 0, tells the Controller to requeue the reconcile key after the Duration.
	// Implies that Requeue is true, there is no need to set Requeue to true at the same time as RequeueAfter.
	RequeueAfter time.Duration
}
```
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Result

(Even that documentation is confusing—the `Reconciler` godoc doesn't mention `RequeueAfter`.  I guess the case could be made that the godoc isn't flat-out _wrong_ as the `RequeueAfter` godoc says that `RequeueAfter` "Implies that Requeue is true".  It's darned confusing though and should be fixed IMO.  Good first issue?)

The operator's `Reconcile` method returns `reconcile.Result{RequeueAfter: e.After()}, nil` if `err` is a `retryable.Error`-type value.  Otherwise, `Reconcile` returns `reconcile.Result{}, err`, which **_still_** causes a retry if `err` is not nil.

Reply:
Anyway, that's all to say that I still believe that returning an error here in case of a conflicting service will cause a bunch of spurious reconciliations.  That doesn't mean that we absolutely cannot use the approach you are using here, especially if the alternative would require significant changes to the operator's design, but I want to make sure we understand the implications of this change, and that the implications aren't _too_ onerous.

Reply:
Hmmm, yea you are right. I missed that, thought it was only based on RequeueAfter. 

So I think we both agree some type of message is appropriate here. What about doing a log.Info() or log.Error() without returning it as an error? Would that be enough?

Do we want intentionally want it to continually try to reconcile? In my testing:
1. I create a conflicting service
2. Create the ingress controller
3. Deleted the conflicting service
4. I found that it eventually reconciles and recreates the service that is now not longer under conflict.

Reply:
@Miciah thoughts?

Reply:
Revisiting this now that I know more about how `watch` in operators work. The issue is: if we don't return an error and reconcile again, it may never reconcile, since deleting a service that isn't owned by the ingresscontroller, will not trigger a reconcilation.

So based on @Miciah's comment "it's better to reconcile more than you need than less than you need", I think the logic should stay the way it is, especially since this is a bit of an edge case. Constant errors make sense because the ingress controller isn't able to do it's job, and justifiably should be complaining often.

Understood the implications.